compiler/rustc_hir_typeck/src/expr.rs

```diff
@@ -1672,7 +1672,17 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
 
         let mut seen_fields = FxHashMap::default();
 
-        let mut error_happened = false;
+        let mut error_happened = if variant.fields.len() > remaining_fields.len() {
+            // Some field is defined more than once. Make sure we don't try to
+            // instantiate this struct in static/const context.
+            let guar = self.dcx().span_delayed_bug(expr.span, "struct fields have non-unique names");
```

Comment:
No need to delay a bug here. Please taint the infcx below when we emit `FieldMultiplySpecifiedInInitializer` or `report_unknown_field`.

Reply:
Actually, I guess I misread the issue. You probably want to do this instead on line `1874` where we call `report_missing_fields`?